packages/nextjs/app/builders/0x981ab0D817710d8FFFC5693383C00D985A3BDa38/page.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import type { NextPage } from "next";
+import { useEnsAddress, useEnsAvatar } from "wagmi";
+import { BlockieAvatar } from "~~/components/scaffold-eth";
+
+const ensName = "zapaz.eth";
+
+const Zapaz: NextPage = () => {
+  const [ensAdddress, setEnsAddress] = useState<string | null>();
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+  const { data: fetchedEnsAddress } = useEnsAddress({ name: ensName, chainId: 1 });
+  const { data: fetchedEnsAvatar } = useEnsAvatar({ name: ensName, chainId: 1 });
```

Comment:
I suggest adding some error handling for each of these with the `error:` prop. If you do, it's nice to have something ready to display to users if the data fetch breaks. If you aren't familiar with that kind of pattern, I'm happy to say more about how you could structure it!

Reply:
added some console.error...

Reply:
I meant displaying something to users if the data fetch fails (which happens more in the real world than you might hope). What could make sense to display as a fallback if there's an error?

Reply:
- as fallback for the ENS name, I thought displaying nothing is ok (there is the address anyway) for both cases :  an fetch error or a address with no ENS...
- as fallback for the ENS avatar, it' not a problem as <Blockie /> already handles it, displaying one `blockie` image if no avatar or error 

don't want to bother user with unuseful error !  (never seen this sort of alert on ENS in Dapps...)
so only alerting in the console

what are you thinking of on your side ?

Reply:
> as fallback for the ENS avatar, it' not a problem as <Blockie /> already handles it, displaying one blockie image if no avatar or error

Yup, SE-2 `BlockieAvatar` blockie image related to address if the `ensImage` passed to is undefined (which means it failed to fail) 

Regarding the use of `useEnsAddress` I think we don't need that hook at all we could just hardcode it since it deterministic / static always

Reply:
> Regarding the use of useEnsAddress I think we don't need that hook at all we could just hardcode it since it deterministic / static always
ok, I'll do like that
in other PR #18 `list of builders` there is a more dynamic version